### DISCUSSION TITLE
Using index on Postgres Full Text Search

### DISCUSSION THREAD

**Discussion by LINK1733** [prisma/prisma#12276] [Q&A] [answered]
So I was looking at the docs for manually making an index when using Full Text Search. I followed the example they have for making the index but when I query with prisma it doesn't seem to be using the index I made. When I query using PGAdmin and tell it to search where to_tsvector('english', indexed column) @@ to_tsquery(search string) it seems to use the index then. Is there something I need to add to prisma to tell it to use the index? Also i removed the data from that tsvector and just replaced it with filler text in place of the columns of my database and what not. Thanks in advance for any advice. If needed, I can provide further details.

**Comment by austincrim** [comment]
Hey @LINK1733 👋 ,

Just to clarify a few things:
- you are following the manual steps outlined [in this doc](https://www.prisma.io/docs/concepts/components/prisma-client/full-text-search#postgresql-1)?
- how are you determining that the Prisma Client is not using the index?

Thanks!

**Reply by LINK1733** [reply]
Yup I followed the steps in that doc you linked. I am determining that it isn't using the index because the search time is taking the same amount as before I created the index.

**Reply by austincrim** [reply]
Would you mind enabling query logging and sharing the generated SQL when running the Prisma query?
```ts
const prisma = new PrismaClient({
  log: ['query'],
})
```

**Reply by LINK1733** [reply]
Ok I have the query logs that it gives me when I run the prisma query at the bottom. I also put the relevant part parts of my schema and the query I am running. Hopefully this helps, if I need to provide more of it I can do that as needed.

```prisma
model igdbGame {
  id                Int                           @id @unique
  name              String
  cover             String?
  parentGame        String?
  versionParent     String?
  summary           String?
  gameListGame      Game[]
  gameName          igdbGameName[]
  gameMode          igdbGameToGameMode[]
  genre             igdbGameToGenre[]
  playerPerspective igdbGameToPlayerPerspective[]
  theme             igdbGameToTheme[]

  @@index([name])
}

model igdbGameName {
  id         String   @id @default(cuid())
  altNameId  Int?     @unique
  name       String
  igdbGameId Int
  igdbGame   igdbGame @relation(fields: [igdbGameId], references: [id], onDelete: Cascade)

  @@unique([altNameId, igdbGameId])
  @@index([name])

}
```

```javascript
let searchResults = await prisma.igdbGameName.findMany({
		take: 30,
		where: {
			name: {
				search: searchInput,
			},
			igdbGame: { versionParent: null },
		},
		orderBy: {
			_relevance: {
				fields: ['name'],
				search: searchInput,
				sort: 'asc',
			},
		},

		select: { igdbGame: { select: { name: true, cover: true, id: true } } },
		distinct: ['igdbGameId'],
	});
```

```sql

prisma:query SELECT "public"."User"."id", "public"."User"."email", "public"."User"."username", "public"."User"."hash" FROM "public"."User" WHERE "public"."User"."id" = $1 LIMIT $2 OFFSET $3

prisma:query SELECT "public"."igdbGameName"."id", "public"."igdbGameName"."igdbGameId" FROM "public"."igdbGameName" WHERE (("public"."igdbGameName"."id") IN (SELECT "t0"."id" FROM "public"."igdbGameName" AS "t0" INNER JOIN "public"."igdbGame" AS "j0" ON ("j0"."id") = ("t0"."igdbGameId") WHERE ("j0"."versionParent" IS NULL AND "t0"."id" IS NOT NULL)) AND to_tsvector(concat_ws(' ', "public"."igdbGameName"."name")) @@ to_tsquery($1)) ORDER BY ts_rank(to_tsvector(concat_ws(' ', "public"."igdbGameName"."name")), to_tsquery($2)) ASC OFFSET $3

prisma:query SELECT "public"."igdbGame"."id", "public"."igdbGame"."name", "public"."igdbGame"."cover" FROM "public"."igdbGame" WHERE "public"."igdbGame"."id" IN ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19,$20,$21,$22,$23,$24,$25,$26,$27,$28,$29,$30) OFFSET $31

```

**Comment by austincrim** [comment, marked as answer]
My colleague pointed out that this is an [outstanding issue](https://github.com/prisma/prisma/issues/8950) we are already investigating. Feel free to track that one as we hope to be working on it ASAP.

**Reply by Claytonn** [reply]
Can we get an update on this? This is a **severe** paint point in my existing project as we're trying to scale. We've waited patiently for almost two years now, it's reaching the point we will need to migrate away from Prisma.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by anuuu16** [comment]
Hello all,
I have prepared a blog for adding **Full-Text Search (FTS) in Prisma with PostgreSQL tsvector, Without Migration Drift**

https://medium.com/@chauhananubhav16/bulletproof-full-text-search-fts-in-prisma-with-postgresql-tsvector-without-migration-drift-c421f63aaab3
